File: lua/hamming/hamming.lua
local hamming = {}

function hamming.compute(s1, s2)
    if #s1 ~= #s2 then return -1 end

    local total = 0
    for i=1,#s1 do
        if s1:byte(i) ~= s2:byte(i) then total = total + 1 end
    end
    return total
end

return hamming

File: ruby/hamming/hamming.rb
module BookKeeping
  VERSION = 2
end

class Hamming
  class << self
    def compute(s1, s2)
      raise ArgumentError unless s1.size == s2.size

      s1.chars.zip(s2.chars).count { |a, b| a != b }
    end
  end
end
